include/drivers/i2c.h

```diff
@@ -553,6 +560,29 @@ static inline int z_impl_i2c_slave_driver_unregister(const struct device *dev)
 	return api->driver_unregister(dev);
 }
 
+#ifdef CONFIG_I2C_GET_CONFIG
+/**
+ * @brief Get configuration of a host controller.
+ *
+ * @param dev Pointer to the device structure for the driver instance.
+ * @param dev_config Pointer to return bit-packed 32-bit value of
+ * the I2C controller configuration.
+ *
+ * @retval 0 If successful.
+ * @retval -EIO General input / output error.
+ * @retval -ERANGE Out of supported i2c frequency.
```

Comment:
This would be clearer as "-ERANGE Configured I2C frequency is invalid."

Reply:
Done